plugins/rest-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ApiSteps.java

```diff
@@ -31,21 +31,25 @@
  */
 public class ApiSteps extends ApiSetupSteps {
 
-    private static ApiSteps instance;
-
     public ApiSteps() {
         initApi();
     }
 
-    public static ApiSteps getInstance() {
-        if (instance == null) {
-            instance = new ApiSteps();
+    static final ThreadLocal<ApiSteps> storage = new ThreadLocal<ApiSteps>() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this anonymous class with a call to "ThreadLocal.withInitial". ([squid:S4065](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS4065))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=126&issues=AWeIHODqcrQ7oSxgxnf-&open=AWeIHODqcrQ7oSxgxnf-)

[ISSUE_KEY]: AWeIHODqcrQ7oSxgxnf-